### PR TITLE
GROW-871: Name and Code are no longer required. 

### PR DESCRIPTION
**Pull request by JordanMJensen**
Connect API will auto-create a code if it is not provided.

### CODE REVIEW

File: vacasa/connect/connect.py [hunk 1]
```diff
@@ -229,6 +227,11 @@ def create_unit(self,
             "secured_by": secured_by,
         }
 
+        if code:
+            payload.update({"code": code})
```
Comment (bijan-vacasa) [inline vacasa/connect/connect.py]:
```suggestion
            payload["code"] = code
```

File: vacasa/connect/connect.py [hunk 2]
```diff
@@ -229,6 +227,11 @@ def create_unit(self,
...
+        if name:
+            payload.update({"name": name})
```
Comment (bijan-vacasa) [inline vacasa/connect/connect.py]:
```suggestion
            payload["name"] = name
```

Comment (JordanMJensen) [reply vacasa/connect/connect.py]: is this a performance thing or a more correct way of doing it?

Comment (bijan-vacasa) [reply vacasa/connect/connect.py]: I doubt it really makes a noticeable difference from a performance perspective, it's more just a readability thing.  And even that is super minor.

**Pull request merged**